Nav bar and footer refactor

mikeploythai opened the pull request:
### PROPOSAL
> Looking at the code for both the nav bar and the footer makes me wanna puke. Not only was it my first foray into web dev when I wrote both of those components, but it was also written with the intent for the site to be standard HTML/CSS. I've gone ahead and rewrote everything to simplify the code and make things look nicer.

#### CHANGELOG
- Both the nav and footer have a breakpoint of 900px (this will become standard across all components)
- Nav bar links are centered to even out the look of the nav
- Theme toggle moved from footer to nav
- Nav bar changes to acmDark in dark mode
- Styling is now hierarchical (ty sass)
- Used Svelte's each block to display links in a more clean fashion
- Fixed horizontal overflow issue caused by footer (this was why `overflow-x: hidden` used to be in the CSS)
- Simplified footer design

This is change #2 of my big update to prepare for next semester's codebase.

Commit:
Refactored nav bar and footer
- Introduced light mode/dark mode SVG icons
- Styled theme toggle and moved it to the nav bar
- CSS was re-written to use hierarchy

Commit:
Changed link spacing to space-evenly

jaasonw committed:
Merge branch 'main' into mike/navbar-footer-refactor

EthanThatOneKid commented on src/lib/components/sections/footer.svelte:
```diff
@@ -1,257 +1,189 @@
+  import YouTube from '../icons/youtube.svelte';
+  import LinkedIn from '../icons/linkedin.svelte';
+
+  const footItem = [
```
Since this variable represents an `Array`, I think this variable name should be plural (e.g. `footerItems`).

EthanThatOneKid replied:
Resolved by <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401/commits/c658547444274150ad65c4fc21a64366bc1b8798>.

EthanThatOneKid committed:
Refactored list of socials into `{#each}` loop

EthanThatOneKid commented on src/lib/components/sections/navbar.svelte:
```diff
@@ -1,18 +1,23 @@
+  onMount(() => (jsEnabled = true));
 
-  const menuItems = [
+  const navItem = [
```
Since this variable represents an `Array`, I think this variable name should be plural (e.g. `navItems`).

EthanThatOneKid replied:
Resolved by <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401/commits/c658547444274150ad65c4fc21a64366bc1b8798>.

EthanThatOneKid commented on src/lib/components/sections/navbar.svelte:
```diff
@@ -1,18 +1,23 @@
 <script lang="ts">
   import { page } from '$app/stores';
+  import LightMode from '../icons/light-mode.svelte';
+  import DarkMode from '../icons/dark-mode.svelte';
```
For the sake of consistency, I think we should import the icons from our internal `/$lib/` folder.

```suggestion
  import LightMode from '$lib/components/icons/light-mode.svelte';
  import DarkMode from '$lib/components/icons/dark-mode.svelte';
```

EthanThatOneKid replied:
Resolved by <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401/commits/084501e0c46803937cc23544bf3c502ff10d47a2>.

EthanThatOneKid committed:
Change paths to string literals

It surprisingly worked, but never mind.

EthanThatOneKid commented on src/lib/components/sections/footer.svelte:
```diff
@@ -1,257 +1,180 @@
+  import Instagram from '../icons/instagram.svelte';
+  import Discord from '../icons/discord.svelte';
+  import YouTube from '../icons/youtube.svelte';
+  import LinkedIn from '../icons/linkedin.svelte';
```
For the sake of consistency, I think we should import the icons from our internal `/$lib/` folder.
```suggestion
  import Instagram from '$lib/components/icons/instagram.svelte';
  import Discord from '$lib/components/icons/discord.svelte';
  import YouTube from '$lib/components/icons/youtube.svelte';
  import LinkedIn from '$lib/components/icons/linkedin.svelte';
```

EthanThatOneKid replied:
Resolved by <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401/commits/084501e0c46803937cc23544bf3c502ff10d47a2>.

EthanThatOneKid commented on src/lib/components/sections/navbar.svelte:
```diff
@@ -43,6 +51,22 @@
+            <span class="dark-toggle"><LightMode /></span>
+          {:else}
+            <span class="dark-toggle"><DarkMode /></span>
+          {/if}
```
Nit
```suggestion
          <span class="dark-toggle">
            {#if $theme === AcmTheme.Dark}
              <LightMode />
            {:else}
              <DarkMode />
            {/if}
          </span>
```

EthanThatOneKid reviewed (changes requested):
Left some minor nitpicks primarily having to do with import paths and variable naming.

EthanThatOneKid committed:
Resolved <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401#discussion_r836995990> and <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401#discussion_r837031236>

EthanThatOneKid committed:
Resolved <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401#discussion_r837032469> and <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401#discussion_r837041513>

Resolved <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401#discussion_r837032469>.
Resolved <https://github.com/EthanThatOneKid/acmcsuf.com/pull/401#discussion_r837041513>.

EthanThatOneKid committed:
Refactored toggle markup

EthanThatOneKid committed:
Merge branch 'mike/navbar-footer-refactor' of https://github.com/EthanThatOneKid/acmcsuf.com into mike/navbar-footer-refactor
